### PR TITLE
Fixes magazine overflow and russian roulette starting with 6(7) bullets

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request
Box magazine code started from ``.len`` instead of ``.len + 1`` when refilling mags which would be fine if BYOND's ``to`` didn't include the last number. Unfortunately, it does, and if the mag has bullets when its refilled it will have one over the limit.

Russian revolver code had a type typo where ammo was added in nonexistent ``/obj/item/ammo_box/magazine/internal/rus357``'s ``Initialize`` instead of ``/obj/item/ammo_box/magazine/internal/cylinder/rus357``. It was pretty clearly was meant to start with a single bullet instead of 6, but neither removing existing ammo nor adding new was implemented properly. I stuff it with 5 blanks instead of making it empty because this will at least allow it to click with current guncode instead of bashing yourself over the head without any feedback, until I rework it.

Related to #84057 but doesn't close it.

## Why It's Good For The Game

Overflow could happen on chaplain's revolver when pray-reloading, giving them one more shot than intended.
Russian revolver is still broken, just slightly less as it now can at least tell you that you failed at killing yourself.

## Changelog
:cl:
fix: Guns no longer can be overfilled by 1 bullet
fix: Russian revolvers now spawn with only 1 live round as originally intended, and click when firing a blank.
/:cl:

### CODE REVIEW

**Comment by YuliceTheDruid** [comment]
Is this just for revolvers or all guns? Because there is in fact a reason most other guns can be "overfilled" by 1; magazines hold the amount and then the chamber itself can hold one bullet. It's why guns IRL tend to have things like "17+1" listed.

**Comment by SmArtKar** [comment]
> Is this just for revolvers or all guns? Because there is in fact a reason most other guns can be "overfilled" by 1; magazines hold the amount and then the chamber itself can hold one bullet. It's why guns IRL tend to have things like "17+1" listed.

Gun code is a mess and there's no differentiation between the two internally, but afaik only revolvers use the code that can trigger this bug. Also this overfills *the magazine*, not the gun itself.

**Comment by uaioy** [comment]
> Is this just for revolvers or all guns? Because there is in fact a reason most other guns can be "overfilled" by 1; magazines hold the amount and then the chamber itself can hold one bullet. It's why guns IRL tend to have things like "17+1" listed.

reminded me how you could load 3 shells on a double barrel shotgun

File: code/modules/projectiles/boxes_magazines/internal/revolver.dm [hunk 1]
```diff
@@ -16,7 +16,10 @@
 	caliber = CALIBER_357
 	max_ammo = 6
 	multiload = FALSE
+	start_empty = TRUE
 
-/obj/item/ammo_box/magazine/internal/rus357/Initialize(mapload)
-	stored_ammo += new ammo_type(src)
+/obj/item/ammo_box/magazine/internal/cylinder/rus357/Initialize(mapload)
 	. = ..()
+	for (var/i in 1 to 5)
```
Comment (Ghommie) [inline code/modules/projectiles/boxes_magazines/internal/revolver.dm:23]:
```suggestion
	for (var/i in 1 to max_ammo - 1)
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Pull request merged**